backend/src/apis/round.rs

```diff
@@ -157,54 +149,26 @@ where
         let mst_inclusion_setup_artifacts: SetupArtifacts =
             generate_setup_artifacts(k, Some(params_path), mst_inclusion_circuit).unwrap();
 
-        let solvency_setup_artifacts_artifacts =
-            generate_setup_artifacts(10, Some(params_path), solvency_circuit).unwrap();
-
-        let trusted_setup = [
-            mst_inclusion_setup_artifacts,
-            solvency_setup_artifacts_artifacts,
-        ];
-
         Ok(Snapshot {
             mst,
             assets_state,
-            trusted_setup,
-        })
-    }
-
-    pub fn generate_proof_of_solvency(&self) -> Result<SolvencyProof, &'static str> {
-        let asset_sums = self
-            .assets_state
-            .iter()
-            .map(|asset| Fp::from_raw(asset.amount.0) as Fp)
-            .collect::<Vec<Fp>>()
-            .try_into()
-            .unwrap();
-        let circuit = SolvencyCircuit::<N_ASSETS, N_BYTES>::init(self.mst.clone(), asset_sums);
-
-        let calldata = gen_proof_solidity_calldata(
-            &self.trusted_setup[1].0,
-            &self.trusted_setup[1].1,
-            circuit,
-        );
-
-        Ok(SolvencyProof {
-            proof_calldata: calldata.0,
-            public_inputs: calldata.1,
+            trusted_setup: mst_inclusion_setup_artifacts,
         })
     }
 
     pub fn generate_proof_of_inclusion(
         &self,
         user_index: usize,
     ) -> Result<MstInclusionProof, &'static str> {
+        let merkle_proof = self.mst.generate_proof(user_index).unwrap();
```

Comment:
As soon as you modify `Snapshot` to take an object with a Binded Trait `Tree` this object will no longer have the `get_entry` method as this method doesn't live on the `Tree` trait. I don't know what's the best way to handle this

Reply:
Well, as you already know I added `get_entry` method in `Tree` trait.
but for removing `get_entry` from `Tree` trait, we have to add `username` and `balances` fields on the `MerkleProof`.
the `entry` usually used to get `username` and `balances`  In the `MstInclusionCircuit`.
What do you think?

Reply:
Ok let's keep it as it is for now! 

If something comes up during the `AggregationMerkleSumTree` implementation, we can just come back to `summa-solvency` and fix that